README.rst

```diff
@@ -0,0 +1,329 @@
+
+The ``PurchaseWorkflow`` is scheduled to execute every hour.
+
+Finally, the application adds a service for querying the results from the Dataset.
+
+Let's take a closer look at the Workflow program.
+
+The ``PurchaseWorkflow`` extends an `AbstractWorkflow
+<http://docs.cdap.io/cdap/current/en/reference-manual/javadocs/co/cask/cdap/api/workflow/AbstractWorkflow.html>`__
+class and overrides the ``configure()`` method:
+
+.. code:: java
+
+  public class PurchaseWorkflow extends AbstractWorkflow {
+    @Override
+    protected void configure() {
+      setName("PurchaseWorkflow");
+      setDescription("Workflow to parse the purchase events and count the revenue per user and per product");
+
+      addMapReduce("PurchaseEventParser");
+
+      condition(new PurchaseEventVerifier())
+        .fork()
+          .addMapReduce("PurchaseCounterByUser")
+        .also()
+          .addMapReduce("PurchaseCounterByProduct")
+        .join()
+      .otherwise()
+        .addAction(new NotifyByEmail())
+      .end();
+    }
+  }
+
+In the ``configure()`` method we specify the topology for connecting the programs which will run as a part of
+the Workflow execution. For ``PurchaseWorkflow``, after setting the name and description for the Workflow, we add
+the MapReduce program ``PurchaseEventParser``. This program will parse raw purchase events and create ``Purchase``
+objects from them.
+
+After that, we add ``condition`` in the Workflow, which takes a predicate ``PurchaseEventVerifier``.
+If the predicate evaluates to true, we ``fork`` the execution of the Workflow into two parallel branches.
+One branch executes the ``PurchaseCounterByUser`` MapReduce program, while the other executes the
+``PurchaseCounterByProduct`` MapReduce program.
+
+If the predicate evaluates to false, then programs added in the ``otherwise`` section will be executed.
+We have added a single dummy custom action, ``EmailNotifier()`` to the ``otherwise`` section as an example
+of what is possible.
+
+Lets take a closer look at the predicate ``PurchaseEventVerifier``.
+
+.. code:: java
+
+  public class PurchaseEventVerifier implements Predicate<WorkflowContext> {
+
+    @Override
+    public boolean apply(WorkflowContext workflowContext) {
+      if (workflowContext == null) {
+        return false;
+      }
+
+      WorkflowToken token = workflowContext.getToken();
+      if (token == null) {
+        return false;
+      }
+
+      Map<String, Map<String, Long>> hadoopCounters = token.getMapReduceCounters();
+      if (hadoopCounters == null) {
+        return false;
+      }
+
+      Map<String, Long> taskCounter = hadoopCounters.get("org.apache.hadoop.mapreduce.TaskCounter");
+
+      if (taskCounter.containsKey("MAP_INPUT_RECORDS")) {
+        long mapInputRecordNumber = taskCounter.get("MAP_INPUT_RECORDS");
+        long mapOutputRecordNumber = taskCounter.get("MAP_OUTPUT_RECORDS");
+        // Return true if at least 80% of the records were processed by previous map job
```

Comment:
all inputs are always processed. I guess you want to say, were successfully parsed ?